### DISCUSSION TITLE
copy/paste/dupilcate shortcuts

### DISCUSSION THREAD

**Discussion by alihammad99** [prevwong/craft.js#701] [Q&A]
I want to ask if copying or duplicating a node (component) using shortcuts like ctr+c / ctrl+v is possible.

I'm trying to make the editor as easy as possible like Figma with basic editing features

**Comment by alihammad99** [comment]
I found a simple solution that could help. However, I still get some type errors but it works:

```
const handleDuplicate = () => {
    // Get the current node
    const node = query.node(id).get();

    if (node) {
      // Create a copy of the node
      const clonedNode = query.createNode(node.data);

      // Insert the cloned node after the current node
      actions.add(clonedNode, node.data.parent, undefined, node.id);
    }
  };
```

**Reply by pcjaarvis** [reply]
This only clones the parent node not its children, any idea how to do that?

**Reply by hugominas** [reply]
check this https://github.com/prevwong/craft.js/issues/316 it copies the node tree
